content/blog/2018-08-21-birds-radolfzell.md

```diff
@@ -0,0 +1,528 @@
+    ##  $ obsDt          : chr  "2018-08-08 13:30" "2018-08-08 13:30" "2018-08-08 13:30" "2018-08-08 13:30" ...
+    ##  $ obsReviewed    : logi  FALSE FALSE FALSE FALSE FALSE FALSE ...
+    ##  $ comName        : chr  "Black-headed Gull" "White Wagtail" "Water Rail" "Tufted Duck" ...
+    ##  $ lat            : num  47.7 47.7 47.7 47.7 47.7 ...
+    ##  $ locID          : chr  "L3314048" "L3314048" "L3314048" "L3314048" ...
+    ##  $ locId          : chr  "L3314048" "L3314048" "L3314048" "L3314048" ...
+    ##  $ howMany        : int  NA 2 1 1 NA 3 NA NA 8 20 ...
+
+Now that we have the occurrence data, let’s plot it to see whether
+trimming is required.
+
+``` r
+crs <- sf::st_crs(landkreis_konstanz)
+
+birds_sf <- sf::st_as_sf(birds,
+                         coords = c("lng", "lat"), 
+                         crs = crs)
+library("ggplot2")
+ggplot() +
+  geom_sf(data = landkreis_konstanz) +
+  geom_sf(data = birds_sf) +
+  theme(legend.position = "bottom") +
+  hrbrthemes::theme_ipsum() +
+  ggtitle("eBird observations over the last 30 days",
+          subtitle = "Observations within a circle around the County of Constance")
+```
+
+![map of raw observations within a
+circle](/img/blog-images/2018-08-21-birds-radolfzell/map_circle-1.png)
+
+Yes, trimming is required! It’d have been too bad not to learn how to do
+it, anyway.
+
+``` r
+in_indices <- sf::st_within(birds_sf, landkreis_konstanz)
+
+trimmed_birds <- dplyr::filter(birds_sf,
+                               lengths(in_indices) > 0)
+
+summarized_birds <- trimmed_birds %>%
+  dplyr::group_by(locName) %>%
+  dplyr::summarise(n = n())
+
+ggplot() +
+  geom_sf(data = landkreis_konstanz) +
+  geom_sf(data = summarized_birds,
+          aes(size = n), show.legend = "point") +
+  hrbrthemes::theme_ipsum() +
+  ggtitle("eBird observations over the last 30 days",
+          subtitle = "County of Constance")
+```
+
+![trimmed observations in the
+county](/img/blog-images/2018-08-21-birds-radolfzell/trimmed1-1.png)
+
+We got 49 observations (`nrow(trimmed_birds)`) of 49 species
+(`length(unique(trimmed_birds$comName))`), over 2 places
+(`length(unique(trimmed_birds$locName))`) during 5 observation sessions.
+Hopefully merely an appetizer to what we can hope to get from using the
+full eBird dataset in the next section…
+
+Before then, let’s just add the MPI on the map.
+
+``` r
+mpi <- opencage::opencage_forward("Am Obstberg 1 78315 Radolfzell", limit = 1)$results
+coords <- data.frame(lon = mpi$geometry.lng,
+                     lat = mpi$geometry.lat)
+crs <- sf::st_crs(landkreis_konstanz)
+
+mpi_sf <- sf::st_as_sf(coords,
+                       coords = c("lon", "lat"), 
+                      crs = crs)
+
+ggplot() +
+  geom_sf(data = landkreis_konstanz) +
+  geom_sf(data = summarized_birds,
+          aes(size = n), show.legend = "point") +
+  geom_sf(data = mpi_sf,
+          shape = 2) +
+  hrbrthemes::theme_ipsum()+
+  ggtitle("eBird observations over the last 30 days",
```

Comment:
this plot is so similar to above plot of obs in last 30 days that I took a while to see the triangle even when it was in the subtitle. Can you do without showing this map?

Reply:
Good idea, I put the MPI on the first map, which makes the code chunk a bit long but I added comments.

I actually think it's quite easy when reading to skip code... I hope I'm not wrong.